vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -17,6 +17,49 @@ def self.table_name
   def index
     redirect_to :action => 'show_list'
   end
+  
+  def scaling
+    assert_privileges("ems_infra_scale")
+    if params[:cancel]
+      redirect_to :action => 'show', :id => params[:id]
+    end
+    
+    drop_breadcrumb(:name => _("Scale Infrastructure Provider"), :url => "/ems_infra/scaling")
+    @infra = EmsOpenstackInfra.find(params[:id])
+    @stack = @infra.orchestration_stacks.first
+    @count_parameters = @stack.parameters.select {|x| x.name.to_s.include?('::count')}
+    
+    if params[:scale]
```

Comment:
can we early exit if !params[:scale] here, please and have one less level of nesting?